### PR TITLE
 For #10465: Add telemetry for voice search 

### PR DESCRIPTION
**Pull request by sblatz**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by sblatz** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**
-How often do users interact with voice search?
-How often do users disable/enable the voice search option in settings?

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**

- We want to know if voice search is useful to users.

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**

* N/A (These are baseline metrics)

4) **Can current instrumentation answer these questions?**
* Currently there are no events to track this event, as the feature is brand new.

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data c](https://wiki.mozilla.org/Firefox/Data_Collection)[ollection ](https://wiki.mozilla.org/Firefox/Data_Collection)[categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the found on the Mozilla wiki.**

* All data is Category 2.

6) **How long will this data be collected?**

Until 09/01/2020

7) **What populations will you measure?**

* All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**

* Glean / Amplitude

9) **Where do you intend to share the results of your analysis?**

* Only on Glean, Amplitude and with mobile teams.

File: app/metrics.yaml [hunk 1]
```diff
@@ -137,6 +137,7 @@ events:
           show_search_shortcuts, open_links_in_a_private_tab (bug in
           implementation https://github.com/mozilla-mobile/fenix/issues/7384),
           pref_key_sync_logins, pref_key_sync_bookmarks, pref_key_sync_history,
+          pref_key_show_voice_search,
```
Comment (liuche) [inline app/metrics.yaml:140]: Hey @sblatz, do you think these pref states should be moved to the metrics ping? These are different, someone actively toggling a pref vs what the value of the pref is - do we care how often a pref is toggled, versus how many people have it enabled/disabled?

Comment (liuche) [inline app/metrics.yaml:140]: Does this not get updated in the metrics.md? I don't see a change in that file included in this PR.

Comment (sblatz) [reply app/metrics.yaml:140]: Unclear. My understanding was that @vesta0 wanted to know when users went in to settings to disable it. 

@vesta0 would you prefer this be sent in the baseline ping? e.g. "user has voice search disabled, has open links in private tab enabled, etc.)? 

I agree this seems more useful but I think we also wanted to know if people were aware they could even *change* these settings.

Comment (sblatz) [reply app/metrics.yaml:140]: Think I just need to do a rebuild of the project to generate. Let's see what Vesta responds to the other comment :)

Comment (sblatz) [reply app/metrics.yaml:140]: As we discussed, let's merge this as is for now. Sounds like there may be some more clean up data science and I need to do on these to get better data from them, but that's a larger project!

**Pull request merged**